ledger/ledger-on-memory/src/test/suite/scala/com/daml/ledger/on/memory/InMemoryLedgerReaderWriterIntegrationSpec.scala

```diff
@@ -44,13 +45,24 @@ abstract class InMemoryLedgerReaderWriterIntegrationSpecBase(enableBatching: Boo
       testId: String,
       metrics: Metrics,
   )(implicit loggingContext: LoggingContext): ResourceOwner[ParticipantState] =
-    new InMemoryLedgerReaderWriter.SingleParticipantOwner(
-      ledgerId,
-      batchingLedgerWriterConfig,
-      participantId,
-      metrics = metrics,
-      engine = Engine.DevEngine()
-    ).map(readerWriter => new KeyValueParticipantState(readerWriter, readerWriter, metrics))
+    new ResourceOwner[ParticipantState] {
```

Comment:
You may want to add a `SingleParticipantOwner` or similar factory function back to `InMemoryLedgerReaderWriter` so that you can reuse the below code in the two relevant integration specs.

Reply:
Done.